spring-web/src/main/java/org/springframework/http/client/HttpComponentsAsyncClientHttpRequestFactory.java

```diff
@@ -117,8 +117,7 @@ public HttpComponentsAsyncClientHttpRequestFactory(
 	 * @since 4.3.10
 	 * @see #setHttpClient(HttpClient)
 	 */
-	public void setAsyncClient(HttpAsyncClient asyncClient) {
-		Assert.notNull(asyncClient, "HttpAsyncClient must not be null");
+	public void setAsyncClient(@Nullable HttpAsyncClient asyncClient) {
```

Comment:
The field should change to not-null along with direct field population in the constructors here. I'll commit this in a bit.